src/main/java/com/google/sps/servlets/YouTubeServlet.java

```diff
@@ -0,0 +1,37 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.Arrays;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.stream.Collectors;
+import com.google.sps.data.Video;
+
+@WebServlet("/yt")
+public class YouTubeServlet extends HttpServlet {
+
+  private Video video;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    String json = gson.toJson(video);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String commentsStr = request.getReader().lines().collect(Collectors.joining());
+    Gson gson = new Gson();
+    String[] commentsArr = gson.fromJson(commentsStr, String[].class);
+    List<String> commentsList = Arrays.asList(commentsArr);
+    video = new Video(commentsList);
+  }
```

Comment:
There is some odd behavior happening in this servlet.

Since video is a private property of the servlet class, you could have a case where user A makes the POST request which populates the video property and then user B makes a GET request which would get the data that user A requested.

GET methods are used to get some kind of information based on the request. POST methods are used to store data that is sent in the request somewhere like in a database. Since you are not getting data from the request that will store somewhere, you don't need a POST method. In this case, the requester gives you some data and is requesting the sentiment score. This seems more like GET.

What I would expect here is have a GET method that takes in the comments from the request, calls the sentiment API, puts the score into the data object, JSONify it, and then send it back.

This change would require client side changes. But like I mention in another comment dont think about the client side code until the back-end is developed and have them as separate PRs. If you are blocking a client side developer from coding, just tell them what the API (emphasis on `interface`) is. They shouldn't care about how back end does the processing.